src/apps/engine/src/main.cpp

```diff
@@ -125,7 +125,10 @@ int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, PSTR szCmdLine,
         MessageBoxA(nullptr, "Another instance is already running!", "Error", MB_ICONERROR);
         return EXIT_SUCCESS;
     }
-    mi_register_output(mimalloc_fun, nullptr);
+    std::string *mi_arg = new std::string("init");
```

Comment:
I don't fully understand the meaning of mi_arg (especially destroy_at for it) 🤔 
if the intention is to remove file at first write, just put a `static bool` and check for it. unlike`std::filesystem::path` it won't be destroyed (because there is nothing to destroy...)